recipes-core/initrdscripts/files/ptest/test_ni_provisioning.sh

```diff
@@ -0,0 +1,176 @@
+	found_error=true
+}
+
+ptest_fail() {
+	echo "FAIL: $PTEST_TEST"
+	exit 1
+}
+
+ptest_pass() {
+	echo "PASS: $PTEST_TEST"
+	exit 0
+}
+
+ptest_skip() {
+	echo "SKIP: $PTEST_TEST"
+	exit 77
+}
+
+
+test_part_id () {
+	local device=${1}
+	local label=${2}
+	local uuid=${3}
+
+	local correct_re_label=${4}
+	local correct_re_uuid=${5}
+
+	echo "test_part_id() ${@:1:3}"
+
+	if [[ ! "${label}" =~ $correct_re_label ]]; then
+		print_error "Partition ${label} @ ${device} does not match required label ${correct_re_label}"
+		found_error=true
+	fi
+	if [[ ! "${uuid}" =~ $correct_re_uuid ]]; then
+		print_error "Partition ${label} @ ${device} does not match required type UUID ${correct_re_uuid}"
+		found_error=true
+	fi
+}
+
+
+found_error=false
+
+if uname --machine | grep -q arm; then
+	echo "INFO: detected an ARM machine architecture." \
+	     "This ptest is only valid for non-ARM, RAUC systems."
+	ptest_skip
+fi
+
+
+## PARTITION TESTS ##
+echo "## Checking that RAUC partitons are present..."
+# The UUIDs here should match those set by the disk_config_x64 file.
+# https://github.com/ni/meta-nilrt/blob/983d50796302b394aa428b992dae48b4da32ff08/recipes-core/initrdscripts/files/disk_config_x64#L28
+RE_ESP_UUID=C12A7328\-F81F\-11D2\-BA4B\-00A0C93EC93B
+RE_X64_ROOT_UUID=4F68BCE3\-E8CD\-4DB1\-96E7\-FBCAF984B709
+
+# Determine the configured /boot storage device. Assert that this device should
+# be the root of the partitions we evaluate in the remainder of this test.
+# For RAUC systems, we expect this to be: /dev/sda, but it isn't required to be.
+boot_part=$(findmnt --kernel --first-only --noheadings --output SOURCE /boot)
+test -n "${boot_part}" || (print_error "No detected /boot mount; system state invalid."; ptest_fail)
+boot_dev=/dev/$(lsblk -no PKNAME ${boot_part})
+test -e "${boot_dev}" || (print_error "Boot device ${boot_dev} does not exist; system state invalid."; ptest_fail)
+
+
+readarray -t parts < <(fdisk -l ${boot_dev} -o Device,Name,Type-UUID | grep -o -E '^\/dev\/.*')
+
+# ASSERT: part 0 is: niboota and is an ESP partition
+test_part_id ${parts[0]} niboota $RE_ESP_UUID
+# ASSERT: part 1 is: nibootb and is an ESP partition
+test_part_id ${parts[1]} nibootb $RE_ESP_UUID
+# ASSERT: part 2 is: niuser and is any partition type
+test_part_id ${parts[2]} niuser $RE_X64_ROOT_UUID
+
+# bail out if the partition tests failed
+$found_error && ptest_fail
+
+
+## RAUC PARTITION DEVICE LINKS TESTING ##
+echo "## Checking for RAUC partition device links in /dev..."
+test -d /dev/niboot || print_error "/dev/niboot does not exist."
+
+niboot_links=(\
+	/dev/niboot/niboot.current \
+	/dev/niboot/niboot.other \
+	/dev/niboot/niboota \
+	/dev/niboot/nibootb \
+	/dev/niboot/niuser \
+)
+for link in ${niboot_links[@]}; do
+	if [ -L "$link" ]; then
+		echo "${link} [OK]"
+	else
+		echo "${link} [FAIL]"
+		print_error "Device link \"$link\" does not exist."
+	fi
+done
+# the link subtests are not critical failures, so keep testing, even if they fail.
```

Comment:
Great, however they are critical - at least to RAUC (cat /etc/rauc/system.conf).  RAUC wouldn't find the slots if these aren't set properly.  rauc-mark-good also uses them.  The initramfs creates the links and uses them itself as part of the boot process, so one could argue that if you're booted and running a ptest, then they're at least good enough, but there's no harm in actually testing and caring about them.

Reply:
@amstewart do you have an upcoming fix for this? If the tests are all going to be automated, I don't see how a test that could error but not fail add value. Noone is probably going to pay attention to the stdout unless there is a failure. So, I'd say either keep and fail the test, or remove.

Reply:
@amstewart and I discussed this, the comment is a little misleading.  The test WILL fail, however the test continues to run the remainder of the tests.  IE The test doesn't bail out just because of this part failing.

Reply:
Does the ptest runner pick up the [FAIL] and ERROR messages to report a failure, or do we need to parse that as a failure when we set up the test runs? I guess I don't understand how the runner reports failures. @amstewart, if you want I can follow up with you offline.